apps/files/src/views/FileReferencePickerElement.vue

```diff
@@ -40,74 +49,55 @@ export default {
 			default: false,
 		},
 	},
-	mounted() {
-		this.openFilePicker()
-		window.addEventListener('click', this.onWindowClick)
-	},
-	beforeDestroy() {
-		window.removeEventListener('click', this.onWindowClick)
+	computed: {
+		containerId() {
+			return `filepicker-${Math.random().toString(36).slice(7)}`
+		},
+		filepickerOptions() {
+			return {
+				allowPickDirectory: false,
+				buttons: this.buttonFactory,
+				container: `#${this.containerId}`,
+				multiselect: false,
+				name: t('files', 'Select file or folder to link to'),
+			}
+		},
 	},
 	methods: {
-		onWindowClick(e) {
-			if (e.target.tagName === 'A' && e.target.classList.contains('oc-dialog-close')) {
-				this.$emit('cancel')
+		t,
+
+		buttonFactory(selected: NcNode[]): IFilePickerButton[] {
+			const buttons = [] as IFilePickerButton[]
+			if (selected.length === 0) {
+				buttons.push({
+					label: t('files', 'Choose file'),
+					type: 'tertiary' as never,
+					callback: this.onClose,
+				})
+			} else {
+				buttons.push({
+					label: t('files', 'Choose {file}', { file: selected[0].basename }),
+					type: 'primary',
+					callback: this.onClose,
+				})
 			}
+			return buttons
 		},
-		async openFilePicker() {
-			OC.dialogs.filepicker(
-				t('files', 'Select file or folder to link to'),
-				(file) => {
-					const client = OC.Files.getClient()
-					client.getFileInfo(file).then((_status, fileInfo) => {
-						this.submit(fileInfo.id)
-					})
-				},
-				false, // multiselect
-				[], // mime filter
-				false, // modal
-				FilePickerType.Choose, // type
-				'',
-				{
-					target: this.$refs.picker,
-				},
-			)
+
+		onClose(nodes?: NcNode[]) {
+			if (!nodes) {
```

Comment:
```suggestion
			if (nodes.length === 0) {
```
Arrays are always truthy

Reply:
Mh, then it should be fixed (also on master), for then it must be true and the other condition could never trigger? Or do I misinterpret something, as you'd wave it through with the checkmark?

Reply:
A prioritized RC3 checkmark rather in anticipation of a quick fix 🙈

Reply:
But it's the last RC before the final, there will be no possibility for a fix anymore, unless something is terribly broken. So either we can live with that, or fix it before merging.

Reply:
Quick fixed :)

Reply:
Thank you! =)

Reply:
Actually the previous quick fix is also not 100% correct as the code reveals some unintuitive behaviour https://github.com/search?q=repo%3Anextcloud-libraries%2Fnextcloud-dialogs%20%22emit(%27close%27%22&type=code which should be handled explicitly

Pushed the real fix

Reply:
In general I'd say we should use `null` rather than `undefined` for explicitness @susnux

Reply:
and the fixes should go to the master PR as well